src/objectExplorer/objectExplorerService.ts

```diff
@@ -89,7 +89,10 @@ export class ObjectExplorerService {
                 }
                 return promise.resolve(self._currentNode);
             } else {
-                // failure
+                // create session failure
+                if (self._currentNode.connectionCredentials.password) {
+                    self._currentNode.connectionCredentials.password = '';
+                }
```

Comment:
What if the user had "save password" selected but the connection failed for some other reason? Wouldn't this reprompt them at that point? That doesn't seem like the correct behavior.